### PR TITLE
PR: Update release instructions to include localization updates

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
+# Release
+
 To release a new version of Spyder you need to follow these steps:
 
+## Localization updates
+
+* Install [gettext-helper](https://github.com/spyder-ide/gettext-helpers) from source.
```
Comment (ccordoba12) [inline RELEASE.md:7]:
```suggestion
* Install [gettext-helpers](https://github.com/spyder-ide/gettext-helpers) from source.
```

File: RELEASE.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+
+* Update `*.pot` and `*.po` files to update translation strings using
```
Comment (ccordoba12) [inline RELEASE.md:9]:
```suggestion
* Update `*.pot` and `*.po` files to update translation strings by following these steps:
```

File: RELEASE.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+  * Run `spyder-gettext scane spyder` to update localization files.
+  * Create and merge a new PR with these updated files
+  * Request translators to update the translation strings on https://crowdin.com/project/spyder
+
+* Close the automatic translation bot PR `translate/<branch-name>` and delete the branch to recreate the branch automatically without conflicts (if any).
```
Comment (ccordoba12) [inline RELEASE.md:14]: Please post below the command to delete the branch.

File: RELEASE.md [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+
+* Wait for the new `translate/<branch-name>` PR to be created by the `spyder-bot`.
+
+* Update `*.mo` files to update translation strings using:
```
Comment (ccordoba12) [inline RELEASE.md:18]:
```suggestion
* Update `*.mo` files to update translation strings by following these steps:
```

File: RELEASE.md [hunk 5]
```diff
@@ -1,5 +1,28 @@
...
+
+* Run `spyder-gettext compile spyder` to update localization binary files.
```
Comment (ccordoba12) [inline RELEASE.md:20]:
```suggestion
  * Run `spyder-gettext compile spyder` to update localization binary files.
```

File: RELEASE.md (hunk 3 above)
Comment (goanpeca) [reply RELEASE.md:14]: I did from the github interface

Comment (goanpeca) [reply RELEASE.md:14]: I can add a comment saying that and https://github.com/spyder-ide/spyder/branches

**Comment by jitseniesen** [comment]
I think it would be helpful to add:
* At which point in the process will the updated strings be available on crowdin?
* At which step do we have to wait for the translators to provide the translations?

**Comment by goanpeca** [comment]
Good point @jitseniesen I will add to it!

**Comment by goanpeca** [comment]
@jitseniesen is it clearer now?

File: RELEASE.md [hunk 6]
```diff
@@ -1,5 +1,32 @@
+# Release
+
 To release a new version of Spyder you need to follow these steps:
 
+## Localization updates
+
+* Install [gettext-helpers](https://github.com/spyder-ide/gettext-helpers) from source.
+
+* Update `*.pot` and `*.po` files to update translation strings by following these steps:
+  * Run `spyder-gettext scane spyder` to update localization files.
```
Comment (jitseniesen) [inline RELEASE.md:10]: `scane` or `scan`?

File: RELEASE.md [hunk 7]
```diff
@@ -1,5 +1,32 @@
...
+  * Using this command: `git push origin --delete translate/<branch-name>`.
+  * Or, on the [github website](https://github.com/spyder-ide/spyder/branches)
+
+* Wait for the new `translate/<branch-name>` PR to be created by the `spyder-bot`. This branch should be created within 10 minutes.
+
+* Update `*.mo` files to update translation strings by following these steps:
+
+  * Run `spyder-gettext compile spyder` to update localization binary files.
+
+* Update the translation PR `translate/<branch-name>` to include these files and squash all commits into 1 single commit. These commits will include the `*.pot`, `*.po` and `*.mo` file changes.
```
Comment (jitseniesen) [inline RELEASE.md:26]: I assume `translate/<branch-name>` needs to be merged into `<branch-name>`. Perhaps should this be mentioned explicitly.

File: RELEASE.md (hunk 6 above)
Comment (goanpeca) [reply RELEASE.md:10]: Typo! `scan`. good catch!

File: RELEASE.md (hunk 7 above)
Comment (goanpeca) [reply RELEASE.md:26]: Agreed!

File: RELEASE.md [hunk 8]
```diff
@@ -1,5 +1,32 @@
...
+  * Using this command: `git push origin --delete translate/<branch-name>`.
+  * Or, on the [github website](https://github.com/spyder-ide/spyder/branches)
+
+* Wait for the new `translate/<branch-name>` PR to be created by the `spyder-bot`. This branch should be created within 10 minutes.
+
+* Update `*.mo` files to update translation strings by following these steps:
+
+  * Run `spyder-gettext compile spyder` to update localization binary files.
+
+* Update the translation PR `translate/<branch-name>` to include these files and squash all commits into 1 single commit. These commits will include the `*.pot`, `*.po` and `*.mo` file changes. Once this has been accomploished merge the PR. Localization updates are now ready.
```
Comment (jitseniesen) [inline RELEASE.md:26]: accomploished -> accomplished

Comment (goanpeca) [reply RELEASE.md]: FixeD!

**Comment by goanpeca** [comment]
Thanks for the review @jitseniesen 

I think this is ready @ccordoba12

**Pull request merged**